hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -209,20 +256,11 @@ std::vector<hardware_interface::StateInterface> GenericSystem::export_state_inte
   }
 
   // Sensor state interfaces
-  for (auto i = 0u; i < info_.sensors.size(); i++)
-  {
-    const auto & sensor = info_.sensors[i];
-    for (const auto & interface : sensor.state_interfaces)
-    {
-      if (!get_interface(
-            sensor.name, sensor_interfaces_, interface.name, i, sensor_states_, state_interfaces))
-      {
-        throw std::runtime_error(
-          "Interface is not found in the standard nor other list. "
-          "This should never happen!");
-      }
-    }
-  }
+  populate_state_interfaces(
+    "standard nor other", info_.sensors, sensor_interfaces_, sensor_states_, state_interfaces);
```

Comment:
is there other possibility than setting this string. What about returning "false" if error happens and then add output here and not in lambda?

Reply:
yes, that can be done. I will change this.